### PR TITLE
Grade Report

### LINKED ISSUE #14: Integrate Grade Report
This is mainly something that I just want to have since I wanted to make my grade report app have a GUI and this could be a good way to do it. I think it could be a useful tool for this productivity suite. Thoughts @ishan0102?

### PR DESCRIPTION
**Pull request by JimothyGreene**
### Problem
We didn't have this widget. Not really a problem, but we wanted to add this widget. 

### Solution
This widget provides an interface to track all of your school assignments. Within this widget there are 4 entities: semesters, courses, assignment types, and assignments. To add a new semester or course, click the Plus button in the bottom right. Upon filling out the form, the new item will appear on the left drawer. Once you select a course, you can then add an assignment type using the same button in the bottom right. Adding a new type will create a table in the main viewport. You can then add an assignment on the bottom row of each table. 

### Testing
I tested this by adding one of each entity on a fresh account. I also clicked on the different components to make sure they all responded appropriately.

### Notes
This PR is massive, so take your time to look through it @ishan0102 . Also, you cannot currently edit or delete any of these items. I wanted to just get this working component pulled in before moving forward with any of that. It should be relatively simple compared to everything that was done for this PR. Also, there are no validations on the frontend for any of the forms. Currently, you just have to make sure you provide all the right information. That is something that also needs to be added.

Also, make sure to run a `docker-compose build` before running this. I added some node modules for the frontend.

Closes #14

### CODE REVIEW

**Comment by JimothyGreene** [comment]
Forgot to mention, but there was no widget on the dashboard that linked to this, and I didn't want to change anything there since the other PR dealt with it. Just go to /grades manually to get to this widget

File: client/src/components/grades/AddCourseForm.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        autoFocus
+        margin="dense"
+        label="Course Name"
+        fullWidth
+        onChange={(e) => {setCourseName(e.target.value)}}
+      />
+      <Typography gutterBottom>
+        Grade Cutoffs
+      </Typography>
+      <Slider
```
Comment (ishan0102) [inline client/src/components/grades/AddCourseForm.js:92]: This is a really nice touch

File: client/src/components/grades/AddCourseForm.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      <TextField
+        autoFocus
+        margin="dense"
+        label="Course Name"
+        fullWidth
+        onChange={(e) => {setCourseName(e.target.value)}}
+      />
+      <Typography gutterBottom>
+        Grade Cutoffs
+      </Typography>
```
Comment (ishan0102) [inline client/src/components/grades/AddCourseForm.js:91]: The slider is a really nice touch

Comment (JimothyGreene) [reply client/src/components/grades/AddCourseForm.js:91]: Yeah it was surprisingly easy to use. Sorts the values for you and everything.

**Review by ishan0102 — APPROVED** [review]
Super solid, definitely one of the coolest PRs to date.

**Pull request merged**